spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -257,6 +317,7 @@ public boolean dropTable(Identifier ident) {
   @Override
   public void renameTable(Identifier from, Identifier to) throws NoSuchTableException, TableAlreadyExistsException {
     try {
+      // can't rename hadoop tables
```

Comment:
This should throw an exception if it can't rename, right?

Reply:
I think this should have a method like `checkNotPathIdentifier` that throws an IllegalArgumentException if a `PathIdentifier` is passed, so that we can ensure that they aren't passed to methods that don't support paths.

Reply:
done